Automattic/jetpack
# AI Featured Image: allow users to type custom prompts on the featured image generation

Fixes #37103.

## Proposed changes:

* Support custom user prompts on the AI Client image generation hook
* Add UI to allow users to type a prompt for a custom image generation

**Expected behavior**

First image:

<img width="600" alt="Screenshot 2024-04-26 at 18 35 22" src="https://github.com/Automattic/jetpack/assets/6760046/9a1b9fbf-5568-4205-8a37-a30164340655">

Asking for a new image with a prompt:

<img width="600" alt="Screenshot 2024-04-26 at 18 35 57" src="https://github.com/Automattic/jetpack/assets/6760046/62a5c4e0-45fc-479a-807d-74c31539ca52">

The result:

<img width="600" alt="Screenshot 2024-04-26 at 18 37 18" src="https://github.com/Automattic/jetpack/assets/6760046/0a6be960-aa8c-45e0-8ef9-e9e3b6d2f9cf">

Asking for a new one:

<img width="600" alt="Screenshot 2024-04-26 at 18 38 19" src="https://github.com/Automattic/jetpack/assets/6760046/4e8a8b49-b47b-4b2f-a8be-4d9356f40759">

The result:

<img width="600" alt="Screenshot 2024-04-26 at 18 38 48" src="https://github.com/Automattic/jetpack/assets/6760046/7dd66933-8572-47f2-be2d-b386bc53ce64">

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor and add some content to it
* Open the Jetpack sidebar and look for the featured image generator button
* Click it to trigger a first generation
* When the image is ready, use the user prompt field to give specific instructions to the generation
* Notice:
   * the UI of the modal, with the user prompt field; is it working properly?
   * the generation behavior; the generation process works? the UI is good while doing it?
   * the quality of the resulting images; the new images are following your instructions?
* Try to set some of the new images as featured and confirm it is set
